src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,272 @@
+        if (!hasActiveSubscription) {
+          if (this.verbose) {
+            logger.logToStderr(`Starting subscription since subscription is not active for the content type`);
+          }
+          let startSubscriptionEndPoint: string = `activity/feed/subscriptions/start?contentType=${(<any>AuditContentTypes)[args.options.contentType]}&PublisherIdentifier=${this.tenantId}`;
+          const requestOptions: any = {
+            url: `${this.serviceUrl}/${this.tenantId}/${startSubscriptionEndPoint}`,
+            headers: {
+              accept: 'application/json;odata.metadata=none'
+            },
+            responseType: 'json'
+          };
+
+          return request.post(requestOptions);
+        }
+        else {
+          return Promise.resolve();
+        }
+      });
+  }
+
+  private getAuditContentList(args: CommandArgs, logger: Logger): Promise<AuditContentList[]> {
+    if (this.verbose) {
+      logger.logToStderr(`Start listing Audit Content URL`);
+    }
+    let SubscriptionListEndpoint: string = `activity/feed/subscriptions/content?contentType=${(<any>AuditContentTypes)[args.options.contentType]}&PublisherIdentifier=${this.tenantId}`;
+
+    if (typeof args.options.startTime !== 'undefined') {
+      const parsedEndDate: any = new Date(args.options.startTime);
+      parsedEndDate.setDate(parsedEndDate.getDate() + 1);
+      SubscriptionListEndpoint += `&starttime=${escape(args.options.startTime)}&endTime=${escape(parsedEndDate.toISOString())}`;
+    }
+    const requestOptions: any = {
+      url: `${this.serviceUrl}/${this.tenantId}/${SubscriptionListEndpoint}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request.get<AuditContentList[]>(requestOptions)
+  }
+
+  private getBatchedPromises(AuditContentLists: AuditContentList[], batchSize: number): void {
+    for (let i: number = 0; i < AuditContentLists.length; i += batchSize) {
+      const PromiseRequestBatch: Promise<AuditlogReport[]>[] = AuditContentLists.slice(i, i + batchSize < AuditContentLists.length ? i + batchSize : AuditContentLists.length).map((AuditContentList: AuditContentList) => {
+        return this.getAuditLogReportforSingleContentURL(AuditContentList.contentUri);
+      })
+      this.batchedPromises.push(PromiseRequestBatch);
+    }
+  }
+
+  private getBatchedAuditlogData(logger: Logger, batchNumber: number, resolve: () => void, reject: (err: any) => void): void {
+
+    if (this.verbose) {
+      logger.logToStderr(`Starting Batch : ${batchNumber}`);
+    }
+    Promise
+      .all(this.batchedPromises[batchNumber])
+      .then((data: any) => {
+        this.completeAuditReports.push(data);
+        if (batchNumber < this.batchedPromises.length - 1) {
+          this.getBatchedAuditlogData(logger, ++batchNumber, resolve, reject)
+        }
+        else {
+          resolve();
+        }
+      }, err => { return Promise.reject(err) });
```

Comment:
I am sorry that it was there in the code base. Since this was relatively complicated code base, I had created a child project to ensure it works as expected. The codes were copied from that and that is why it had some of the unwanted codes which should have been cleansed before I initiated the PR. Sorry that you had to go through all those. Totally appreciate your patience

I have removed the `err `snippet since we may not need since that would already be handled in the root method.

Reply:
Removing `err` leads to a possible unhandled promise. If any of the promises will fail, we'll end up with a potentially unhandled exception. The right way to handle it would be to have the `getBatchedAuditlogData` method take `reject` callback as another argument and in case of error in `Promise.all` call `reject` passing the error.

Reply:
Thanks for the recommendation on that. Totally make sense to have `err` handling which is needed when you are batching promises. 

I have modified the code and have added the error handling section and have added the code which will reject the `err`. [Line 197](https://github.com/arjunumenon/cli-microsoft365/blob/18c5824a71523bb97acc33b4f173d9c2f3d2d5bd/src/m365/tenant/commands/auditlog/auditlog-report.ts#L197) to 199.